### ISSUE TITLE
`netlify lm:setup` crashes

### ISSUE THREAD

**Issue by ehmicky** [netlify/cli#514]
**- Do you want to request a *feature* or report a *bug*?** 
Bug

**- What is the current behavior?**
`netlify lm:setup` crashes.

**- If the current behavior is a bug, please provide the steps to reproduce.**
I ran the following commands in order:
  - `apt-get install git-lfs`
  - `git lfs version`
  - `netlify plugins:install netlify-lm-plugin`
  - `netlify lm:install`
  - `netlify lm:setup`

All commands worked except the last one which shows:

```
Logging into your Netlify account...
Opening https://app.netlify.com/authorize?response_type=ticket&ticket=xxxxxx
---------------------------
Error: Unable to open browser automatically
spawn /home/ether/.local/share/netlify-cli/node_modules/netlify-lm-plugin/node_modules/@netlify/cli-utils/node_modules/cli-ux/lib/xdg-open ENOENT

Please open your browser & open the URL below to login:
https://app.netlify.com/authorize?response_type=ticket&ticket=xxxxxx
---------------------------
TypeError: Cannot read property 'slug' of undefined
    at LmSetup.authenticate (~/.local/share/netlify-cli/node_modules/netlify-lm-plugin/node_modules/@netlify/cli-utils/src/index.js:116:21)
    at async LmSetup.run (~/.local/share/netlify-cli/node_modules/netlify-lm-plugin/lib/commands/lm/setup.js:24:29)
    at async LmSetup._run (~/.local/share/netlify-cli/node_modules/netlify-lm-plugin/node_modules/@netlify/cli-utils/node_modules/@oclif/command/lib/command.js:34:20)
    at async Config.runCommand (~/.nvm/versions/node/v12.10.0/lib/node_modules/netlify-cli/node_modules/@oclif/config/lib/config.js:151:9)
    at async Main.run (~/.nvm/versions/node/v12.10.0/lib/node_modules/netlify-cli/node_modules/@oclif/command/lib/main.js:21:9)
    at async Main._run (~/.nvm/versions/node/v12.10.0/lib/node_modules/netlify-cli/node_modules/@oclif/command/lib/command.js:44:20)
```

**- What is the expected behavior?**
`netlify lm:setup` should not crash

**- Local Environment Information**
Paste the results of `netlify status --verbose` here

```
──────────────────────┐
 Current Netlify User │
──────────────────────┘
Name:  ehmicky
Email: [edited]
Teams: 
  ehmicky's team: Owner Collaborator Controller

────────────────────┐
 Environment Info   │
────────────────────┘

  System:
    OS: Linux 5.0 Ubuntu 19.04 (Disco Dingo)
    CPU: (16) x64 Intel(R) Core(TM) i9-9900K CPU @ 3.60GHz
  Binaries:
    Node: 12.10.0 - ~/.nvm/versions/node/v12.10.0/bin/node
    npm: 6.11.3 - ~/.nvm/versions/node/v12.10.0/bin/npm
  Browsers:
    Chrome: 76.0.3809.132
    Firefox: 69.0

────────────────────┐
 Netlify Site Info  │
────────────────────┘
Current site: [edited]
Netlify TOML: /home/ether/code/cv-website/netlify.toml
Admin URL:    https://app.netlify.com/sites/[edited]
Site URL:     [edited]
Site Id:      418b94bc-93cd-411a-937a-ae4c734f17c4

```

**Comment by daixiang0** [comment]
@ehmicky @erezrokah is there any workaround or solution for this? I met this too.

**Comment by erezrokah** [comment]
Sorry @daixiang0, but we haven't been able to look into this yet.
Also not sure if this should be handled in the CLI or in https://github.com/netlify/netlify-lm-plugin

**Comment by naavis** [comment]
I am having the same problem on Windows and Powershell. Running `netlify lm:setup` fails with `TypeError: Cannot read property 'slug' of undefined` after authorizing the application in my browser. I am trying to migrate from using Github LFS to Netlify Large Media, if that plays a role in this.

**Comment by erezrokah** [comment]
Thanks for the additional information @naavis. We're in the [process](https://github.com/netlify/cli/pull/1489) of moving https://github.com/netlify/netlify-lm-plugin into the CLI repo. We'll consider this issue during the migration

**Comment by erwinkarim** [comment]
i'm seeing this exact same problem on macOS 11

**Comment by MCJack123** [comment]
I'm experiencing this on both macOS 11 and Ubuntu 18.04. I'm using Node v14.x on both machines. However, I've examined this a bit and it appears this issue could be caused by a backend issue with the API.

The line causing the error as referenced in the OP (mine doesn't show the stack trace) points to calling the `listAccountsForUser` endpoint. It then looks for an account in the returned list that has `type` set to `PERSONAL`. I tried accessing this API endpoint for my account with `netlify api`, but it appears the `type` parameter is missing in the response:

```
$ netlify api listAccountsForUser | grep type
    "type_name": "Starter",
    "type_id": "5d2464889e975c00bab54e55",
    "type_slug": "starter",
```

[The docs](https://open-api.netlify.com/#operation/listAccountsForUser) state that `type` should be defined, but it isn't in this response.

This doesn't excuse the fact that the script assumes it found an account without checking the result. A more user-friendly approach would be to check that `account` isn't `null`, and if it is print an actual error about that, instead of continuing and throwing a cryptic exception.

**Comment by eduardoboucas** [comment]
We've been able to reproduce the issue and we're working on a fix as we speak. We'll update this issue as soon as it's fixed.

Thanks for your patience and for the detailed information! 🙌🏻

**Comment by erezrokah** [comment]
Sorry for the trouble everyone.


```sh
cd ~/.local/share/netlify-cli
npm i @netlify/cli-utils@latest 
# yarn also works
yarn add @netlify/cli-utils@latest
```

**Comment by erezrokah** [comment]
ile

**Comment by erezrokah** [comment]
Sorry for the trouble everyone.

While we figure this out can you try the following workaround

```sh
cd ~/.local/share/netlify-cli
# on windows this should be `cd %LOCALAPPDATA%\netlify-cli`

npm i @netlify/cli-utils@latest 
# yarn also works
yarn add @netlify/cli-utils@latest
```

**Comment by raphink** [comment]
@erezrokah

Ubuntu 20.10 here.

After updating to `@netlify/cli-utils@1.1.0` as you suggested, I don't get the warning about opening the authorization URL anymore, but I still get the `slug` error:

```
❯ npm i @netlify/cli-utils@latest

npm WARN rm not removing /home/raphink/.local/share/netlify-cli/node_modules/.bin/semver as it wasn't installed by /home/raphink/.local/share/netlify-cli/node_modules/semver
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN listr-update-renderer@0.5.0 requires a peer of listr@^0.14.2 but none is installed. You must install peer dependencies yourself.

+ @netlify/cli-utils@1.1.0
added 37 packages from 6 contributors, removed 262 packages, updated 365 packages and audited 403 packages in 32.39s

37 packages are looking for funding
  run `npm fund` for details

found 4 low severity vulnerabilities
  run `npm audit fix` to fix them, or `npm audit` for details


   ╭───────────────────────────────────────────────────────────────╮
   │                                                               │
   │      New major version of npm available! 6.14.8 → 7.5.2       │
   │   Changelog: https://github.com/npm/cli/releases/tag/v7.5.2   │
   │               Run npm install -g npm to update!               │
   │                                                               │
   ╰───────────────────────────────────────────────────────────────╯

❯ netlify lm:setup
  ✔ Checking Git version [2.27.0]
  ✔ Checking Git LFS version [2.11.0]
  ✔ Checking Git LFS filters
  ✔ Installing Netlify's Git Credential Helper for Linux
  ✔ Configuring Git to use Netlify's Git Credential Helper
Logging into your Netlify account...
Opening https://app.netlify.com/authorize?response_type=ticket&ticket=4eb1a000a94e630bb619ef25139e5ee7
    TypeError: Cannot read property 'slug' of undefined
```

**Comment by erezrokah** [comment]
Hi @raphink, can you verify you ran that command inside `~/.local/share/netlify-cli`? That directory is used to share dependencies for the CLI plugins

**Comment by raphink** [comment]
Yes, I did, and this is what `netlify` resolves to:

```
❯ pwd
/home/raphink/.local/share/netlify-cli
❯ which netlify
netlify: aliased to /usr/local/lib/node_modules/netlify-cli/bin/run
```

**Comment by erezrokah** [comment]
@raphink yep, that doesn't look like. Let me see if I can find the correct place to force an update to that dependency.

**Comment by raphink** [comment]
FYI, I just tried forcing it system wide with `sudo`, and it didn't work either.

**Comment by overlordofmu** [comment]
I believe some people have reported using an auth token copied from the web UI resolved the issue for them:

https://docs.netlify.com/cli/get-started/#obtain-a-token-in-the-netlify-ui

**Comment by erezrokah** [comment]
Thanks @overlordofmu I totally overlook that angle. That should work

**Comment by overlordofmu** [comment]
It is only a workaround but it does get some people "working" again while a fix in the CLI tool is being developed.

**Comment by jameshfisher** [comment]
> I believe some people have reported using an auth token copied from the web UI resolved the issue for them:
> 
> https://docs.netlify.com/cli/get-started/#obtain-a-token-in-the-netlify-ui

Hey @overlordofmu, could you expand on this? Setting this env var does not make any difference for me:

```
$ NETLIFY_AUTH_TOKEN=my_auth_token netlify lm:setup
Logging into your Netlify account...
Opening https://app.netlify.com/authorize?response_type=ticket&ticket=06249518ca4b75ce7d4a663dc0b4cf2b
    TypeError: Cannot read property 'slug' of undefined
```

Is this what I should be doing, or is the workaround different somehow?

**Comment by jameshfisher** [comment]
> I believe some people have reported using an auth token copied from the web UI resolved the issue for them:
> 
> https://docs.netlify.com/cli/get-started/#obtain-a-token-in-the-netlify-ui

Hey @overlordofmu, could you expand on this? Setting this env var does not make any difference for me:

```
$ NETLIFY_AUTH_TOKEN=my_auth_token netlify lm:setup
Logging into your Netlify account...
Opening https://app.netlify.com/authorize?response_type=ticket&ticket=qwertyuiopkjhgfdsazxcvbnm
    TypeError: Cannot read property 'slug' of undefined
```

Is this what I should be doing, or is the workaround different somehow?

**Comment by uncasually** [comment]
Hey everyone

Found a workaround that worked for me on Mac.

- added `NETLIFY_AUTH_TOKEN` as and environment variable to terminal
- since I was having issues with npm, I ran it in yarn `yarn add @netlify/cli-utils@latest` in `~/.local/share/netlify-cli`
- after that, navigated back to the already linked folder  and ran `netlify lm:setup` and it FINALLY worked.

I suspect for me is because I am behind some versions on npm maybe since running `npm i -g netlify-cli@latest` resulted in some npm warnings.

Hope it helps some of you.
